### DISCUSSION TITLE
Custom Tinker's materials are now viable, in case nobody noticed

### DISCUSSION THREAD

**Discussion by pm065** [NillerMedDild/Enigmatica6#2294] [General]
Of course, this might have been discussed on `hell` I mean Discord. I have no way to know :p (technically I do but I never want to go there ever again) (last time I used discord it was horrible)
But in case nobody noticed at all...
Kube now supports custom fluids! https://mods.latvian.dev/books/kubejs/page/custom-fluids
(immediately begins branch to add molten metals for materials)

**Comment by pm065** [comment]
Don't worry, I haven't forgotten. The fluids are done, but I'm taking the opportunity to do a bunch more Tinker's things (;

**Reply by pm065** [reply]
One of them (multiple maybe?) may be a surprise, but hopefully pleasant one(s) :smile:
Also, these 'bunch more things' are not actual material JSONs (I do not feel confident balancing things)

**Reply by pm065** [reply]
I intend to create a series of Tinker's PRs in this order:
- [ ] Tinker's Foundation - The base of the other Tinker's expansions. Nearly finished! It contains the metal fluids in a new fluid registry startup script (finished), the two seecrets (one is finished, one is in progress) and a couple of misc tweaks, mainly related to quests
- [ ] Tinker's Unification - Integrate the fluids into the unification script
- [ ] Tinker's Expertise - Tweaks to Expert Mode relating to TiC. Most will be to do with Create's later portions since a) it's the tech mod I like most and am most familiar with, and b) I think it fits very well with Tinker's.
- [ ] Tinker's Jewels - Add gemstone fluids (which aren't nearly as important as metals, since they aren't used for tools)
- [ ] Tinker's Addendum - Add several misc/mod compat things like Molten Blizz/Basalz/Blitz and a slightly more efficient recipe for Andesite Alloy using the Smeltery.

And then the actual tool materials are somebody else's problem.

I might also need a few custom textures for some of the things _hint hint Tinker's Delight hint hint_

**Reply by NielsPilgaard** [reply]
Cool! Looking forward to seeing those PRs. And making materials I guess :P

**Reply by pm065** [reply]
Oh, yes, and keep in mind there will likely be an extremely large amount of errors and mistakes since Foundation will be (already is!) my largest PR yet (:

**Reply by pm065** [reply]
Hm, we have a slight problem. FTB Ultimine **completely** invalidates the Tinker's Broad Axe (tree chopper) and Vein Hammer (vein miner), and there's no way to remove tool types :/. I've added a disclaimer in their quests (fine, I give in, one of the non-liquid things was a quest chapter for TiC :smile:) but it still feels really off.

**Reply by NielsPilgaard** [reply]
Ultimine takes a large toll on ones hunger though, and I think many would prefer using tools in stead of Ultimine, it feels better imo :P

**Reply by pm065** [reply]
Oh, yeah, I forgot about hunger loss from Ultimine

**Reply by pm065** [reply]
Here you go, spoiler thing (still open to changes ofc)

![image](https://user-images.githubusercontent.com/81559564/120119432-8de4d380-c18f-11eb-9c84-b0686ed1c0d0.png)
As you can see, not all the TiC tools have yet been implemented, and in fact some like Arrows, Bolts, Longswords, Rapiers and Longbows will not be returning. Despite this, like Blood Magic, the mod is perfectly playable even in alpha.
Other notable changes include:

- The Melter has been moved to base TiC and is now required to make the Smeltery Controller (cover a seared brick in copper liquid)
- Nether Grout (We are Grout!) has been added, along with Scorched Bricks and the Foundry multiblock which specialises in higher capacity and more efficient item melting but it can't alloy (a Nether Melter variant is planned)
- Eventually, there will be end-themed multiblocks too that specialize in gas manipulation, presumably with End Grout (You are Grout?) and end bricks
- Stencil Table has been removed (part stencil types are also gone, reduced to atoms but for some reason you still need blank patterns)
- Number of parts has been reduced massively to prevent redundancy, the aim is for all parts to be used in at least two tools
- Temporary casts are now made of sand instead of clay
- The Tool Station is now the Tinker Station and now only has 3 slots, the Tool Forge is now the Tinker's Anvil and has the original 5 slots (and the ability to create heavy tools)
- Slime and Islands have been reworked massively

Just in case you were unaware of any of these 1.16 changes (:
(That paper thing is a perpetually-hidden todo for upcoming Tinker's features)

**Reply by pm065** [reply]
Here's all the liquids (top lot is kubejs, bottom tinker's defaults)
I dare you to guess them :smile:
![image](https://user-images.githubusercontent.com/81559564/120158759-62dc9d00-c1ec-11eb-8a4f-51cac93baa09.png)

**Reply by pm065** [reply]
There's a slight consistency issue with the bucket textures, I think that may be resolvable though
EDIT: Sadly not, actually, until Kube allows custom bucket textures

**Reply by pm065** [reply]
Ah yes, ingots of honey
![image](https://user-images.githubusercontent.com/81559564/120159360-09c13900-c1ed-11eb-989e-2b92458b8b30.png)

**Comment by NielsPilgaard** [comment]
At this point I think you should give Discord another try so you can join the dev chat :P

**Reply by pm065** [reply]
Really, really sorry, but I am absolutely adamant that I won't and nothing will change my mind, I hope this doesn't come across as rude :/

**Reply by NielsPilgaard** [reply]
No worries, it doesn't come across as rude :)

**Comment by pm065** [comment]
Okay, I'm incredibly sorry, but I don't feel like I'm enjoying this anymore, and I did something incredibly thoughtless and stupid which means I now feel tired AND guilty... Could these things wait until after 5.0? The vital things like trees and crops are here: https://gist.github.com/pm065/fecf831ca12c406e737902775b198ff0
I couldn't figure out the Akashic Tome, sorry ):

**Reply by NielsPilgaard** [reply]
Totally understandable, I hope you don't beat yourself up about it!
Most of the stuff can easily wait until after 0.5.0, we may implement some of it in the mean time though :)
